### DISCUSSION TITLE
BRAT Summaries and Visualizations

### DISCUSSION THREAD

**Discussion by joewheaton** [Riverscapes/riverscapes-tools#442] [BRAT]
Just moving an email conversation over here as it is quite helpful to get us thinking about our HTML reports more explicilty.

@philipbaileynar posted to @AldenShallcross:

> Joe forward me your spreadsheet and document with some BRAT summary charts, asking how easy it would be to include them in each BRAT run. The answer is VERY easy. Take a look,
> 
> https://youtu.be/LV0W8ZBKZiA
> 
> As I say in the video; I welcome any and all suggestions for improvements in how we can communicate the results of all our tools.

[![Video Link](http://img.youtube.com/vi/LV0W8ZBKZiA/0.jpg)](http://www.youtube.com/watch?v=LV0W8ZBKZiA "")

Then @AldenShallcross suggested:

> I dig it.  This is great and directly aligns with the agreement’s objective of creating information and knowledge from data….as well as incorporating this info into decisions/workflows.  My guess is that the tools are also “value added” features that would bring people back again and again to a platform, tool, etc…….and that they make the info/knowledge that can be gained from the riverscapes data available to a much broader user base (NGO’s, watershed groups, those who can’t use GIS or wouldn’t take the time to run the calculations manually, etc.). 

> What about providing a user the ability to produce the report for a user-defined area of interest?  For example, we may want to view the summaries for just BLM lands within a specific field office or district.  Therefore, we would like to be able to provide a mask/ROI for BLM lands, calculate the stats for just that area, and output a summary report. 
 
[fwdbratresultsmontanadakotasquestionaboutslope.zip](https://github.com/Riverscapes/riverscapes-tools/files/7245832/fwdbratresultsmontanadakotasquestionaboutslope.zip)

**Comment by joewheaton** [comment]
I love this stuff @philipbaileynar and @AldenShallcross.


A few things from an example pyBRAT project:  http://brat.riverscapes.xyz/Regions/USA/IDFG_Idaho/

As we discussed @philipbaileynar, the pyBRAT had a [`Collect_Summary_Products.py`](https://github.com/Riverscapes/pyBRAT/blob/master/Collect_Summary_Products.py) that assembled the following (typically at HUC 8, but we also combined all of these for Region or Statewide summaries.

## Tabular Summaries
 For a given area of interest, we typically summarize BRAT by: 


Segment   Type | Stream Length (m) | Stream Length (Km) | Stream Length (mi) | % of Drainage Network | Surveyed Dams Count | BRAT Estimated Capacity Dam   Count | Average Surveyed Dam Density   (Dams/Km)
-- | -- | -- | -- | -- | -- | -- | --

The `BRAT Estimated CApacity Dam Count` is the calculated density multiplied by segment length summed for entire AOI.


If we are lucky enough to have some dam counts (e.g. from Dam Surveys at desktop or field), we can then additionally calculate:


Surveyed   Dams Count |  Average Surveyed Dam Density   (Dams/Km) |  % Modeled Capacity | Electivity Index
-- | -- | -- | --

Together these look like:


![image](https://user-images.githubusercontent.com/9723905/135126181-7bef2fde-8b1e-4507-af5e-906a59cf665b.png)

An example is below for the Middle Bear HUC 8:
[MiddleBear_SummaryTables_Perennial.xlsx](https://github.com/Riverscapes/riverscapes-tools/files/7245882/MiddleBear_SummaryTables_Perennial.xlsx)

Notice there are the following tabular Summaries by primary outputs:

#### WAtershed Summary
![image](https://user-images.githubusercontent.com/9723905/135138692-7046ab85-0d2c-4177-aeb2-b9ef3fd13bab.png)

#### Density Correlations
![image](https://user-images.githubusercontent.com/9723905/135138593-e9d6dee7-292f-498b-96ff-7bcf8b89c3da.png)

#### Existing Dam Building Capacity
![image](https://user-images.githubusercontent.com/9723905/135138545-36bc394f-d009-4b06-9416-4ac2fecdf101.png)


#### Existing Dam Complex Size
![image](https://user-images.githubusercontent.com/9723905/135138455-e0608fa6-00a1-404f-bc57-0f2917a4c529.png)

#### Historic Dam Building Capacity
![image](https://user-images.githubusercontent.com/9723905/135138421-3cc7c146-04d9-48ad-8be9-2a85a1c46bd9.png)

#### Historic Dam Complex Size
![image](https://user-images.githubusercontent.com/9723905/135138385-b1f6d6e9-4df5-41e1-8931-49b8ba5f914e.png)

#### Existing vs. Historic Capacity
![image](https://user-images.githubusercontent.com/9723905/135138338-c9118efd-ab3c-4b66-ad57-aaf233fea4c3.png)

#### Conservation Restoration
![image](https://user-images.githubusercontent.com/9723905/135138164-a915f0ae-1395-4db2-8ba8-1a8799ca7264.png)

#### Unsuitable or Limited
![image](https://user-images.githubusercontent.com/9723905/135138108-1f4a1675-01b8-4052-bb71-61ace0caf707.png)

#### Undesireable Dams
![image](https://user-images.githubusercontent.com/9723905/135138027-4e4a8ad0-ba9a-40b3-a992-c4433da6db5a.png)

#### Management Strategies
![image](https://user-images.githubusercontent.com/9723905/135137987-2cb3d33b-37cc-44f5-98b2-91c69c9a9101.png)

#### Predicted vs. Surveyed
![image](https://user-images.githubusercontent.com/9723905/135137946-db354b29-deb6-40b8-9e43-933b0697a6ca.png)

#### Electivity Index
![image](https://user-images.githubusercontent.com/9723905/135137901-1d7ef6f2-28e4-4425-a7a2-f0a182d621d3.png)

### A Statewide Summary Example
This runs same stuff, but summing all of the other models (i.e. 80+ HUC 8s here)
[Idaho_SummaryTable.xlsx](https://github.com/Riverscapes/riverscapes-tools/files/7245883/Idaho_SummaryTable.xlsx)


### A simple comparison of Existing & Historic

![image](https://user-images.githubusercontent.com/9723905/135125374-640bc108-8627-4a3c-9a9a-4cda4cd6a094.png)

### Use Legends to Color Bar Plots
Although the bar plot above is not a good example, our preferred plots are horizontal bar plots (there is a guy from EPA who gave a talk here a few years back and talked about why they are preferable). I can't find the papers, but the take homes were:
- Easier to interpret than pie charts (especially when more categories)
- Richer information (you can put two scales (e.g. percent of network, miles of network, etc.) and labels.

Here's an example:
![image](https://user-images.githubusercontent.com/9723905/135125802-b0229c88-8e06-42b1-9015-a8ac7b872fed.png)

**Reply by philipbaileynar** [reply]
- nearly everything you suggest is easy. Just bums on seats. 
- electivity and other analysis that use dam observations will have to wait until those data are either in a brat project or available via API from other sources. They are not currently readily available.
- I prefer bar over pie. If we must use pie charts then let's use donut pie. Tufte has good explanation for this.
